staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go

```diff
@@ -64,7 +66,7 @@ func ServeHTTPWithETag(
 	responsewriters.WriteObjectNegotiated(
 		serializer,
 		DiscoveryEndpointRestrictions,
-		AggregatedDiscoveryGV,
+		targetGV,
```

Comment:
naively, I would expect this to be the only point where we must do conversion.

Reply:
Here + `kube-aggregator/pkg/apiserver/handler_discovery.go` to be able to fetch v2beta1 discovery from aggregated apiservers.